### PR TITLE
Allow REINDEX TABLE to work on dist hypertables

### PR DESCRIPTION
**Pull request by pmwkaa**
Issue #3018

### CODE REVIEW

File: tsl/test/sql/dist_ddl.sql [hunk 1]
```diff
@@ -134,6 +133,10 @@ DROP TABLE disttable, disttable;
 
 \set ON_ERROR_STOP 1
 
+REINDEX TABLE disttable;
+REINDEX TABLE CONCURRENTLY disttable;
+REINDEX (VERBOSE) TABLE disttable;
```
Comment (svenklemm) [inline tsl/test/sql/dist_ddl.sql:138]: Can we have a check here that the reindex actually happened.

**Review by nikkhils — APPROVED** [review]
After Sven's comment is addressed, it should be good to go.

**Comment by pmwkaa** [comment]
@svenklemm, @nikkhils any ideas how to actually test it? We have similar test for single node, but I think it only tests that the command returns VERBOSE output which could not be done here. Is there are a way to check that the command took effect on the index/table using pg catalog, stats etc?

**Comment by svenklemm** [comment]
> @svenklemm, @nikkhils any ideas how to actually test it? We have similar test for single node, but I think it only tests that the command returns VERBOSE output which could not be done here. Is there are a way to check that the command took effect on the index/table using pg catalog, stats etc?

Reindex writes a new file so you can check with pg_relation_filepath that the path for an index is different before/after operation.

**Review by mkindahl — APPROVED** [review]
LGTM.

**Comment by mkindahl** [comment]
> LGTM.

You should probably have a better message for the commit though, this is very terse.

**Pull request merged**